python/lsst/ts/ATWhiteLightSource/wlsCSC.py

```diff
@@ -292,30 +378,33 @@ async def reconnect_kiloarc_or_fault(self, max_attempts=3):
             num_attempts = 0
             while num_attempts < max_attempts:
                 num_attempts += 1
-                self.log.debug("iteration "+str(num_attempts))
+                self.log.debug("iteration " + str(num_attempts))
                 try:
                     self.log.debug("trying reconnect...")
                     self.kiloarcModel.component.reconnect()
                     self.kiloarcModel.component.checkStatus()  # this triggers the exception we're fishing for.
                     self.log.debug("it worked")
                     break
                 except ConnectionException:
-                    self.log.debug("kiloarc connection problem, attempting reconnect " + str(num_attempts))
+                    self.log.debug(
+                        "kiloarc connection problem, attempting reconnect "
+                        + str(num_attempts)
+                    )
                     self.log.debug(str(num_attempts) + " " + str(max_attempts))
                     if num_attempts >= max_attempts:
                         self.log.debug("going FAULT")
-                        self.summary_state = salobj.State.FAULT
+                        self.fault()
```

Comment:
You need to pass `code` and `report` to `self.fault`. Although this works with salobj 5 it will stop working with salobj 6.

Reply:
saving this for the next ticket

Reply:
uhm.. this one is pretty straightforward.. just do 

```
MAX_ATTEMPTS_ERROR = 100
```

at the top and then

```
self.fault(code=MAX_ATTEMPTS_ERROR, report=f"Max attempts ({max_attempts}) to ??? exceeded.")
```